File: sample/src/main/java/com/schibsted/spain/barista/sample/PreferencesActivity.java
package com.schibsted.spain.barista.sample;

import android.app.Activity;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;
import android.widget.TextView;

public class PreferencesActivity extends Activity {

  private static final String PREFERENCE_KEY = "value";
  private static final int PREFERENCE_DEFAULT_VALUE = 0;

  private SharedPreferences preferences;
  private TextView currentValueText;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_preferences);

    preferences = PreferenceManager.getDefaultSharedPreferences(this);

    currentValueText = ((TextView) findViewById(R.id.preference_current_value));
    findViewById(R.id.preference_increment_button).setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View v) {
        incrementValue();
        showCurrentValue();
      }
    });

    showCurrentValue();
  }

  private void showCurrentValue() {
    int currentValue = preferences.getInt(PREFERENCE_KEY, PREFERENCE_DEFAULT_VALUE);
    currentValueText.setText(String.valueOf(currentValue));
  }

  private void incrementValue() {
    int currentValue = preferences.getInt(PREFERENCE_KEY, PREFERENCE_DEFAULT_VALUE);
    int newValue = currentValue + 1;
    preferences.edit()
        .putInt(PREFERENCE_KEY, newValue)
        .apply();
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/PreferencesActivity.kt
package com.schibsted.spain.barista.sample

import android.app.Activity
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import android.view.View
import android.widget.TextView

class PreferencesActivity : Activity() {
  private var preferences: SharedPreferences? = null
  private var currentValueText: TextView? = null
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_preferences)
    preferences = PreferenceManager.getDefaultSharedPreferences(this)
    currentValueText = findViewById<View>(R.id.preference_current_value) as TextView
    findViewById<View>(R.id.preference_increment_button).setOnClickListener {
      incrementValue()
      showCurrentValue()
    }
    showCurrentValue()
  }

  private fun showCurrentValue() {
    val currentValue = preferences!!.getInt(PREFERENCE_KEY, PREFERENCE_DEFAULT_VALUE)
    currentValueText!!.text = currentValue.toString()
  }

  private fun incrementValue() {
    val currentValue = preferences!!.getInt(PREFERENCE_KEY, PREFERENCE_DEFAULT_VALUE)
    val newValue = currentValue + 1
    preferences!!.edit()
      .putInt(PREFERENCE_KEY, newValue)
      .apply()
  }

  companion object {
    private const val PREFERENCE_KEY = "value"
    private const val PREFERENCE_DEFAULT_VALUE = 0
  }
}